packages/typescript-estree/src/removeParserOptionsThatPromptTypechecking.ts

```diff
@@ -0,0 +1,18 @@
+import { TSESTreeOptions } from './parser-options';
+
+/**
+ * Removes options that prompt the parser to parse the project with type
+ * information. In other words, you can use this if you are invoking the parser
+ * directly, to ensure that one file will be parsed in isolation, which is much,
+ * much faster.
+ *
+ * @see https://github.com/typescript-eslint/typescript-eslint/issues/8428
+ */
+export function removeParserOptionsThatPromptTypechecking(
+  opts: TSESTreeOptions,
+): TSESTreeOptions {
+  delete opts.EXPERIMENTAL_useProjectService;
+  delete opts.project;
+
+  return opts;
```

Comment:
I went with `delete`, since `eslint-plugin-import` is the main target (and they clone the options prior to passing them), but parameter mutation is probably quite surprising for other users. 

I considered rest/spread, but I am not sure whether it intersects with`eslint-plugin-import`'s support target; if I understand correctly, they are quite strict with older node support. I'll take any and all insights 😁

Reply:
`eslint-plugin-import` is an ...unusual project 😄. They have much older semantics & support targets than what we support. Let's not worry about their old targets. They'll have to conditionally import this util anyway (https://github.com/import-js/eslint-plugin-import/pull/2963/files#r1485454097).

+1 to `...` spreading and not mutating parameters. Agreed that parameter mutation is probably quite surprising for other users!